ci/scripts/r_sanitize.sh

```diff
@@ -36,34 +36,36 @@ ncores=$(${R_BIN} -s -e 'cat(parallel::detectCores())')
 echo "MAKEFLAGS=-j${ncores}" >> ${rhome}/etc/Renviron.site
 
 # build first so that any stray compiled files in r/src are ignored
-${R_BIN} CMD build .
-${R_BIN} CMD INSTALL ${INSTALL_ARGS} arrow*.tar.gz
+${R_BIN} CMD build --no-build-vignettes --no-manual .
 
 # But unset the env var so that it doesn't cause us to run extra dev tests
 unset ARROW_R_DEV
 
 # Set the testthat output to be verbose for easier debugging
 export ARROW_R_VERBOSE_TEST=TRUE
+export _R_CHECK_FORCE_SUGGESTS_=FALSE
```

Comment:
Since we are using check, but we also are pruning dependencies like duckdb, we need this. We could also _not_ prune dependencies which would be how CRAN runs these, but we've never have sanitizer issues on our pruned dependency tests (most of which skip on cran too) so I'm not super worried. It would be easy to add them later.